### ISSUE TITLE
using static does not work with private type

### ISSUE THREAD

**Issue by YairHalberstadt** [dotnet/roslyn#42567]
**Version Used**: master as of 10th march

**Steps to Reproduce**:

```csharp
using static Outer.Inner;

public class Outer
{
    
    public void M() {
        var x = Prop;
    }
    private static class Inner
    {
        public static int Prop => 42;
    }
}
```

**Expected Behavior**: No Errors

**Actual Behavior**: 
error CS0122: 'Outer.Inner' is inaccessible due to its protection level
error CS0103: The name 'Prop' does not exist in the current context

I think this is a bug because the [Spec](https://github.com/dotnet/csharplang/blob/master/spec/namespaces.md#using-static-directives) makes no mention of such restrictions.

**Comment by YairHalberstadt** [comment]
According to @AlekseyTs this is to spec: See his comments at https://github.com/dotnet/csharplang/pull/3299.
